pymeasure/instruments/hp/hp663X2A.py

```diff
@@ -0,0 +1,233 @@
+    output_enabled = Instrument.control(
+        "OUTP?", "OUTP %s",
+        """Control Enable[Disable] the source output""",
+        validator=strict_discrete_set,
+        values=BOOL_MAPPINGS,
+        map_values=True,
+    )
+
+    voltage_setpoint = Instrument.control(
+        "VOLT?", "VOLT %s",
+        """Control output voltage setpoint""",
+    )
+
+    voltage_trigger_level = Instrument.control(
+        "VOLT:TRIG?", "VOLT:TRIG %s",
+        """Control the pending triggered voltage level""",
+    )
+
+    voltage_protection = Instrument.control(
+        "VOLT:PROT?", "VOLT:PROT %s",
+        """Control the voltage protecton limit
+        Accepts floats or LEV MAX""",
+    )
+
+    current_limit = Instrument.control(
+        "CURR?", "CURR %s",
+        """Control Output current limit setpoint""",
+    )
+
+    current_trigger_level = Instrument.control(
+        "CURR:TRIG?", "CURR:TRIG %s",
+        """Control the pending triggered current limit""",
+    )
+
+    current_protection_enabled = Instrument.control(
+        "CURR:PROT:STAT?", "CURR:PROT:STAT %s",
+        """Control current protecton.
+        Disables output if overcurrent is tripped. Off on reset""",
+        validator=strict_discrete_set,
+        values=BOOL_MAPPINGS,
+        map_values=True,
+    )
+
+    # #Measurement commands
+    current = Instrument.measurement(
+        "MEAS:CURR?",
+        """Measure the DC output current.""",
+    )
+
+    current_rms = Instrument.measurement(
+        "MEAS:CURR:ACDC?",
+        """Measure the AC+DC RMS output current.""",
+    )
+
+    current_min = Instrument.measurement(
+        "MEAS:CURR:MIN?",
+        """Measure the minimum DC output current from a sample.""",
+    )
+
+    current_max = Instrument.measurement(
+        "MEAS:CURR:MAX?",
+        """Measure the maximum DC output current from a sample.""",
+    )
+
+    voltage = Instrument.measurement(
+        "MEAS:VOLT?",
+        """Measure the dc output voltage.""",
+    )
+
+    voltage_rms = Instrument.measurement(
+        "MEAS:VOLT:ACDC?",
+        """Measure the ac+dc RMs output voltage.""",
+    )
+
+    voltage_min = Instrument.measurement(
+        "MEAS:VOLT:MIN?",
+        """Measure the low voltage level from a sample.""",
+    )
+
+    voltage_max = Instrument.measurement(
+        "MEAS:VOLT:MAX?",
+        """Measure the high voltage level from a sample.""",
+    )
+
+    # Output Trigger related commands
+    def init_single_output_trigger(self):
+        """Set the state of the triggering system to "wait-for-trigger".
+
+        The source will change the output levels to voltage_trigger_level and current_trigger_level.
+        It will return to Idle after a single trigger"""
+        self.write("INIT:NAME TRAN"),
+
+    def enable_cont_output_trigger(self):
+        """Set the state of the triggering system to "wait-for-trigger".
+
+        The source will change the output levels to voltage_trigger_level and current_trigger_level.
```

Comment:
If you want to link to attributes, you can use this:
```suggestion
        The source will change the output levels to :attr:`voltage_trigger_level` and :attr:`current_trigger_level`.
```

~~~:meth:`name_of_method`~~~ refers to a method...